check_api/src/main/java/com/google/errorprone/ErrorPronePlugins.java

```diff
@@ -45,11 +54,50 @@ public static ScannerSupplier loadPlugins(ScannerSupplier scannerSupplier, Conte
       return scannerSupplier;
     }
     ClassLoader loader = fileManager.getClassLoader(StandardLocation.ANNOTATION_PROCESSOR_PATH);
-    Iterable<BugChecker> extraBugCheckers = ServiceLoader.load(BugChecker.class, loader);
-    if (Iterables.isEmpty(extraBugCheckers)) {
+
+    List<Class<? extends BugChecker>> bugCheckerClasses;
+    // Synchronize to prevent races
+    synchronized(ErrorPronePlugins.class) {
+      // disable URL caching to avoid stale state bugs when run within a daemon-building process
```

Comment:
Can you also provide a test or at least a repro so we can make sure the fix sticks?

And javac uses `ServiceLoader` to load `com.sun.source.util.Plugin`s. Do you know if there's something specific to Error Prone's use of service loading that's causing this issue? If not, it seems like it might be a more general problem that needs to be addressed or worked around in Buck?